src/pocketmine/Player.php

```diff
@@ -2468,7 +2469,23 @@ public function handleInventoryTransaction(InventoryTransactionPacket $packet) :
 
 				switch($type){
 					case InventoryTransactionPacket::USE_ITEM_ON_ENTITY_ACTION_INTERACT:
-						break; //TODO
+						$item = $packet->trData->itemInHand;
+						$clickPos = $packet->trData->clickPos;
+						$slot = $packet->trData->hotbarSlot;
```

Comment:
These should not be used. The client can lie about these and you're not checking if they are valid. If this was an attack the client could say it had a diamond sword and the server wouldn't know any better.

Reply:
I'll change L#2472 to use the server-side Item in hand on the next commit...
clickPos will still remain dependent on the client as they may be using 'split touch'.

Reply:
Yes, clickpos is fine. I was referring to the slot and item. Sorry for not being clear.